File: python/tests/power/ina219_test.py
import time
import _testconfig as cfg
from periph.transport.i2c_micropython import I2CTransport
from periph.chips.power.ina219 import INA219Full

from machine import I2C, Pin

passed = 0
failed = 0


def check_true(label, condition):
    global passed, failed
    if condition:
        print('PASS', label)
        passed += 1
    else:
        print('FAIL', label)
        failed += 1


i2c = I2C(cfg.I2C_ID, sda=Pin(cfg.SDA), scl=Pin(cfg.SCL), freq=cfg.FREQ)
transport = I2CTransport(i2c, cfg.ADDR)
ina = INA219Full(transport)

check_true('voltage non-negative', ina.voltage()       >= 0.0)
check_true('shunt_voltage finite', ina.shunt_voltage() > -1.0)
check_true('current finite',       ina.current()       > -10.0)
check_true('power non-negative',   ina.power()         >= 0.0)

check_true('conversion_ready', ina.conversion_ready())
check_true('no overflow',      not ina.overflow())

ina.configure(1, 3, 3, 3, 7)
check_true('voltage after configure', ina.voltage() >= 0.0)

ina.shutdown()
time.sleep_ms(1)
ina.wake()
check_true('wake: voltage non-negative', ina.voltage() >= 0.0)

ina.reset()
check_true('reset: voltage non-negative', ina.voltage() >= 0.0)

print('===DONE: {} passed, {} failed==='.format(passed, failed))

File: nodejs/tests/power/ina219_test.js
'use strict';

const { I2CConnection } = require('../../packages/periph/src/connection/i2c');
const { INA219Full }   = require('../../packages/periph/src/chips/power/ina219');

const I2C_BUS  = parseInt(process.env.I2C_BUS  || '1',  10);
const I2C_ADDR = parseInt(process.env.I2C_ADDR  || '0x40', 16);

let passed = 0;
let failed = 0;

function checkTrue(label, condition) {
    if (condition) { console.log('PASS', label); passed++; }
    else           { console.log('FAIL', label); failed++; }
}

async function main() {
    const connection = new I2CConnection(I2C_BUS, I2C_ADDR);
    const ina = new INA219Full(connection);

    checkTrue('voltage non-negative', (await ina.voltage())      >= 0.0);
    checkTrue('shunt_voltage finite', (await ina.shuntVoltage()) > -1.0);
    checkTrue('current finite',        (await ina.current())      > -10.0);
    checkTrue('power non-negative',   (await ina.power())        >= 0.0);

    checkTrue('conversion_ready', await ina.conversionReady());
    checkTrue('no overflow',      !(await ina.overflow()));

    await ina.configure(1, 3, 0x03, 0x03, 7);
    checkTrue('after configure: voltage non-negative', (await ina.voltage()) >= 0.0);

    await ina.shutdown();
    const end = Date.now() + 1;
    while (Date.now() < end) {}
    await ina.wake();
    checkTrue('wake: voltage non-negative', (await ina.voltage()) >= 0.0);

    await ina.reset();
    checkTrue('after reset: voltage non-negative', (await ina.voltage()) >= 0.0);

    await connection.close();

    console.log(`===DONE: ${passed} passed, ${failed} failed===`);
    process.exit(failed === 0 ? 0 : 1);
}

main();
